client/mm/libxc/mexc.go

```diff
@@ -241,19 +240,17 @@ func (b *mexcOrderBook) convertBookUpdate(update *mxctypes.BookUpdate) (bids, as
 }
 
 // Connect implements the dex.Connector interface.
-// Synchronizes orderbook: buffer WS updates, fetch REST snapshot, apply buffered updates.
+// Synchronizes orderbook: fetch REST snapshot, then accept fresh WebSocket updates.
 func (b *mexcOrderBook) Connect(ctx context.Context) (*sync.WaitGroup, error) {
 	// Synchronization variables
 	var syncMtx sync.Mutex
-	var syncCache []*mxctypes.BookUpdate
 
 	resyncChan := make(chan struct{}, 1)
 
 	desync := func(resync bool) {
-		// Clear the sync cache and trigger a book refresh.
+		// Mark as unsynced and trigger a book refresh.
 		syncMtx.Lock()
 		defer syncMtx.Unlock()
-		syncCache = make([]*mxctypes.BookUpdate, 0)
 		if b.synced.Load() {
 			b.synced.Store(false)
```

Comment:
This can be atomic using "Swap", currently it is not really as there is a gap between checking and setting. https://pkg.go.dev/sync/atomic#Bool.Swap